### ISSUE TITLE
Is it  possible to run tests using Gauge daemon?

### ISSUE THREAD

**Issue by apssouza22** [getgauge/gauge#1684]
Hi, I have been searching for this answer but I could not found anything in the documentation.
I am using my the Golang plugging and I would like to run my tests from my code and I thought using daemon could be an option. 
My language plugin can connect to the Gauge daemon but it doesn't send any message to start the tests. Is it possible to fire the `Gauge run tests` through the gauge daemon? If it is possible where can I get instruction to do so?

Thanks

**Comment by BugDiver** [comment]
Hi, I'm not sure if I'm understanding your use case correctly. But running the tests from a test itself will not be possible via gauge.
Maybe you could explain your use case and scenario a bit so that we can understand it better.

**Comment by apssouza22** [comment]
Hi @BugDiver. Basically I am trying to see how I could add debugging support(add breakpoints) when running the tests. I thought if I could start the Golang tests(connection to the Gauge server) from my IDE it would work. Do you have any suggestion? How is the debugging support for the other languages?

**Comment by sriv** [comment]
tldr; - this is best done in the gauge-go runner, rather than your test code.

debugging support for other languages is done via the language runner. Some language runtimes allow attaching a debugger. Ex: `gauge-java` uses this as an argument to the `java` command:

```
-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=<PORT>,timeout=25000
```

gauge-js uses `node-inspector` - see [`debugging.md`](https://github.com/getgauge/gauge-js/blob/c70223639f5fb15b6d7d1f7fe5a39d814570eac2/docs/debugging.md) for details. 

python uses `ptvsd` but there is a pull request open to move this to `debugpy`.

For golang, I think the best approach would be for `gauge-go` to include support for attaching a debugger like `delve`.

**Comment by apssouza22** [comment]
Thanks @sriv  It helped a lot!
